generators/php/codegen/src/context/AbstractPhpGeneratorContext.ts

```diff
@@ -218,6 +245,115 @@ export abstract class AbstractPhpGeneratorContext<
         }
     }
 
+    public dereferenceOptional(typeReference: TypeReference): TypeReference {
+        if (typeReference.type === "container" && typeReference.container.type === "optional") {
+            return typeReference.container.optional._visit({
+                container: (container) => {
+                    return this.dereferenceOptional(TypeReference.container(container));
+                },
+                named: (named) => {
+                    return this.dereferenceOptional(TypeReference.named(named));
+                },
+                primitive: (primitive) => {
+                    return TypeReference.primitive(primitive) as TypeReference;
+                },
+                unknown: () => {
+                    return TypeReference.unknown() as TypeReference;
+                },
+                _other: () => {
+                    return TypeReference.unknown() as TypeReference;
+                }
+            });
```

Comment:
nit: This is fine as-is, but I think we can just handle this similar to the `isOptional` helper - special case the `optional` case, but return the `TypeReference` directly in all other cases. That way we don't need to cast anything `as TypeReference` at all.

Reply:
done